_pytest/pytester.py

```diff
@@ -418,16 +418,8 @@ class Testdir:
     def __init__(self, request, tmpdir_factory):
         self.request = request
         self._mod_collections = WeakKeyDictionary()
-        # XXX remove duplication with tmpdir plugin
-        basetmp = tmpdir_factory.ensuretemp("testdir")
         name = request.function.__name__
-        for i in range(100):
-            try:
-                tmpdir = basetmp.mkdir(name + str(i))
-            except py.error.EEXIST:
-                continue
-            break
-        self.tmpdir = tmpdir
+        self.tmpdir = tmpdir_factory.mktemp(name, numbered=True)
```

Comment:
my current understanding is, that this needs the keep parameter, or it will delete test folders

Reply:
Not sure, this is exactly what `tmpdir` itself [does](https://github.com/pytest-dev/pytest/blob/master/_pytest/tmpdir.py#L125), and it does keep folders around.

Reply:
Default keep parameter is 3. So I guess the three last folders are being kept?

Reply:
actually keep is set to 0 so it should be perfectly fine - it may be advisable to factor out that function in a future pr

sorry for the noise